hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractBulkDelete.java

```diff
@@ -0,0 +1,185 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.fs.contract.s3a;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.BulkDelete;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.contract.AbstractContractBulkDeleteTest;
+import org.apache.hadoop.fs.contract.AbstractFSContract;
+import org.apache.hadoop.fs.s3a.Constants;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
+import org.apache.hadoop.fs.s3a.S3ATestUtils;
+import org.apache.hadoop.fs.statistics.MeanStatistic;
+import org.apache.hadoop.io.wrappedio.WrappedIO;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.skip;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.touch;
+import static org.apache.hadoop.fs.s3a.S3ATestUtils.createFiles;
+import static org.apache.hadoop.fs.statistics.IOStatisticAssertions.lookupMeanStatistic;
+import static org.apache.hadoop.fs.statistics.StoreStatisticNames.STORE_IO_RATE_LIMITED_DURATION;
+import static org.apache.hadoop.fs.statistics.StoreStatisticNames.SUFFIX_MEAN;
+import static org.apache.hadoop.io.wrappedio.WrappedIO.bulkDelete;
+import static org.apache.hadoop.test.LambdaTestUtils.intercept;
+
+/**
+ * Contract tests for bulk delete operation for S3A Implementation.
+ */
+@RunWith(Parameterized.class)
+public class ITestS3AContractBulkDelete extends AbstractContractBulkDeleteTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ITestS3AContractBulkDelete.class);
+
+    /**
+     * Delete Page size: {@value}.
+     * This is the default page size for bulk delete operation for this contract test.
+     * All the tests in this class should pass number of paths equal to or less than
+     * this page size during the bulk delete operation.
+     */
+    private static final int DELETE_PAGE_SIZE = 20;
+
+    private final boolean enableMultiObjectDelete;
+
+    @Parameterized.Parameters(name = "enableMultiObjectDelete = {0}")
+    public static Iterable<Object[]> enableMultiObjectDelete() {
+        return Arrays.asList(new Object[][] {
+                {true},
+                {false}
+        });
+    }
+
+    public ITestS3AContractBulkDelete(boolean enableMultiObjectDelete) {
+        this.enableMultiObjectDelete = enableMultiObjectDelete;
+    }
+
+    @Override
+    protected Configuration createConfiguration() {
+        Configuration conf = super.createConfiguration();
+        S3ATestUtils.disableFilesystemCaching(conf);
+        S3ATestUtils.removeBaseAndBucketOverrides(conf,
```

Comment:
I think if bulk delete is disabled for a bucket, we should just skip the bulk delete set of tests. otherwise test runs against GCS will fail.